src/main/java/duke/customClass/LogicHandler.java

```diff
@@ -0,0 +1,152 @@
+package duke.customClass;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.util.List;
+
+/**
+ * Handles the process that happens when based on the input of the user.
+ */
+public class LogicHandler {
+    /**
+     * Triggered by the user typing 'list'.
+     * Prints out all tasks in the list currently.
+     * @param list List of tasks.
+     */
+    public void list(List<Task> list) {
+        // Check if the command is list and display the list of duke.tasks.
+        String temp = "";
+
+        for (int i = 0; i < list.size(); i++) {
+            temp += String.format("%d. %s", i + 1, list.get(i));
+            if (i != list.size() - 1) {
+                temp += "\n";
+            }
+        }
+        System.out.println(temp);
+    }
```

Comment:
Perhaps a more intuitive method name that describes what is being operated should be used since `list` is generic and could refer to other listings like help options. I noticed the same in several other files as well so it might be good to re-look at class methods that might not be clear in conveying what they do :)